vpr/src/base/read_place.cpp

```diff
@@ -318,15 +319,21 @@ void print_place(const char* net_file,
 
     fp = fopen(place_file, "w");
 
-    fprintf(fp, "Netlist_File: %s Netlist_ID: %s\n",
-            net_file,
-            net_id);
-    fprintf(fp, "Array size: %zu x %zu logic blocks\n\n", device_ctx.grid.width(), device_ctx.grid.height());
-    fprintf(fp, "#block name\tx\ty\tsubblk\tlayer\tblock number\n");
-    fprintf(fp, "#----------\t--\t--\t------\t-----\t------------\n");
+    if (is_place_file) {
+        fprintf(fp, "Netlist_File: %s Netlist_ID: %s\n",
+                net_file,
+                net_id);
+        fprintf(fp, "Array size: %zu x %zu logic blocks\n\n", device_ctx.grid.width(), device_ctx.grid.height());
+        fprintf(fp, "#block name\tx\ty\tsubblk\tlayer\tblock number\n");
+        fprintf(fp, "#----------\t--\t--\t------\t-----\t------------\n");
+    }
 
     if (!place_ctx.block_locs.empty()) { //Only if placement exists
         for (auto blk_id : cluster_ctx.clb_nlist.blocks()) {
+            // if block is not placed, skip (useful for printing legalizer output)
+            if (!is_place_file && (place_ctx.block_locs[blk_id].loc.x == -1)) {
```

Comment:
I suggest moving 
static constexpr int INVALID_X = -1;

from initial_placement.cpp to vpr_types.h (right above the definition of t_place_loc, and adding a comment saying blocks that are unplaced/have an invalid placement can test for this value. 
Then change the == -1 test to == INVALID_X